test/integration/examples/apiserver_test.go

```diff
@@ -655,6 +606,224 @@ func TestAggregatedAPIServerRejectRedirectResponse(t *testing.T) {
 	}
 }
 
+func prepareAggregatedWardleAPIServer(ctx context.Context, t *testing.T, namespace string, requestHeadersUIDOverride []string) (*kastesting.TestServer, *sampleserver.WardleServerOptions, int) {
```

Comment:
Lets split the test refactor out into its own PR so we can merge that first and make this change easier to review.

Reply:
opened https://github.com/kubernetes/kubernetes/pull/126270 for that